### PR TITLE
Blueprint to Import Postgres DB data into RDS

### PR DESCRIPTION
**Pull request by akankshakumari393**
## Change Overview
This PR adds an example blueprint to import On premise database into RDS Database. This blueprint would take the backup as usual and during restore it would fetch instance details from configmap and would perform restore on the specified host.

*Prerequisite*:
RDS instance would be present before performing restore

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [x] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

- [x] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

* Manually Create RDS instance on AWS, make sure it is accessible from outside.

* Install postgres application and create Test database and add some data into that
```
helm install my-release --create-namespace --namespace postgres-test bitnami/postgresql --set image.registry=ghcr.io --set image.tag=v9.99.9-dev --set image.repository=kanisterio/postgresql

kubectl exec -ti my-release-postgresql-0 -n postgres-test -- bash
postgres@my-release-postgresql-0:/$ PGPASSWORD=${POSTGRES_PASSWORD} psql -U postgres
psql (14.0)
Type "help" for help.

postgres=# CREATE DATABASE test;
CREATE DATABASE
postgres=# \c test
You are now connected to database "test" as user "postgres".
test=# CREATE TABLE COMPANY(
test(#      ID INT PRIMARY KEY     NOT NULL,
test(#      NAME           TEXT    NOT NULL,
test(#      AGE            INT     NOT NULL,
test(#      ADDRESS        CHAR(50),
test(#      SALARY         REAL,
test(#      CREATED_AT    TIMESTAMP
test(# );
CREATE TABLE
test=# INSERT INTO COMPANY (ID,NAME,AGE,ADDRESS,SALARY,CREATED_AT) VALUES (10, 'Paul', 32, 'California', 20000.00, now());

INSERT 0 1
test=# 
test=# 
test=# INSERT INTO COMPANY (ID,NAME,AGE,ADDRESS,SALARY,CREATED_AT) VALUES (20, 'Omkar', 32, 'California', 20000.00, now());

INSERT 0 1
test=# 
test=# 
test=# 
test=# select * from company;
 id | name  | age |                      address                       | salary |         created_at         
----+-------+-----+----------------------------------------------------+--------+----------------------------
 10 | Paul  |  32 | California                                         |  20000 | 2022-11-30 11:28:08.528807
 20 | Omkar |  32 | California                                         |  20000 | 2022-11-30 11:28:20.556397
(2 rows)



```
* Install Kanister
```
helm repo add kanister https://charts.kanister.io
helm install kanister --namespace kanister --create-namespace kanister/kanister-operator --set image.tag=0.84.0
```
* create blueprint
```
kubectl create -f ./examples/postgresql/rds-postgres-blueprint.yaml -n kanister
```
* create Profile 

* Create backup actionset
```
kanctl create actionset --action backup --namespace kanister --blueprint rds-postgres-bp --statefulset postgres-test/my-release-postgresql --profile postgres-test/s3-profile-hknjg
```
this would run backup action

Before Restore Create config map and secret, that contains the details of RDS instance.

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: dbconfig
data:
  postgres_host: <instance host name>
---

apiVersion: v1
kind: Secret
metadata:
  name: dbsecret
stringData:
  postgres_username: <database_username>
  postgres_password: <database_password>

```
* create restore action
```
kanctl --namespace kanister create actionset --action restore --from backup-tkjpp --config-maps dbconfig=postgres-test/dbconfig --secrets dbsecret=postgres-test/dbsecret
```
Once restore action is successful verify the data in RDS instance

```
PGPASSWORD="secret99" psql --host akanksha-rds-test.cjbctojw4ahh.us-west-2.rds.amazonaws.com -U postgres -d postgres -p 5432
```
```
psql (12.12 (Ubuntu 12.12-0ubuntu0.20.04.1), server 14.5)
WARNING: psql major version 12, server major version 14.
         Some psql features might not work.
SSL connection (protocol: TLSv1.2, cipher: ECDHE-RSA-AES256-GCM-SHA384, bits: 256, compression: off)
Type "help" for help.

postgres=> \c test
psql (12.12 (Ubuntu 12.12-0ubuntu0.20.04.1), server 14.5)
WARNING: psql major version 12, server major version 14.
         Some psql features might not work.
SSL connection (protocol: TLSv1.2, cipher: ECDHE-RSA-AES256-GCM-SHA384, bits: 256, compression: off)
You are now connected to database "test" as user "postgres".
test=> select * from company;

id | name  | age |                      address                       | salary |         created_at         
----+-------+-----+----------------------------------------------------+--------+----------------------------
 10 | Paul  |  32 | California                                         |  20000 | 2022-11-30 11:28:08.528807
 20 | Omkar |  32 | California                                         |  20000 | 2022-11-30 11:28:20.556397
(2 rows)

```

### CODE REVIEW

File: examples/postgres-RDS/README.md [hunk 1]
```diff
@@ -0,0 +1,296 @@
+# Postgres-RDS
+
+[PostgreSQL](https://www.postgresql.org/) is an object-relational database management system (ORDBMS) with an emphasis on extensibility and on standards-compliance.
+
+## Introduction
+
+This example is to demonstrate how Kanister can be used to import data in AWS RDS instance from On-prem postgres using Kanister functions
```
Comment (viveksinghggits) [inline examples/postgres-RDS/README.md:7]:
```suggestion
This example is to demonstrate how Kanister can be used to import data in AWS RDS instance from On-prem Postgres using Kanister functions
```
postgres -> Postgres

File: examples/postgres-RDS/README.md [hunk 2]
```diff
@@ -0,0 +1,296 @@
+# Postgres-RDS
+
+[PostgreSQL](https://www.postgresql.org/) is an object-relational database management system (ORDBMS) with an emphasis on extensibility and on standards-compliance.
```
Comment (viveksinghggits) [inline examples/postgres-RDS/README.md:3]: what is on standards-compliance?

File: examples/postgres-RDS/README.md [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+- Kanister controller version 0.84.0 installed in your cluster
+- Kanctl CLI installed (https://docs.kanister.io/tooling.html#kanctl)
+
+## Create RDS instance on AWS
+
+We need to have a postgres RDS instance where we would restore the data.
+
+> You can skip this step if you already have an RDS instance created
+
+RDS instance needs to be reachable from outside world. So make sure that you have VPC with security group having rule to allow ingress traffic on 5432 TCP port.
```
Comment (viveksinghggits) [inline examples/postgres-RDS/README.md:22]:
```suggestion
RDS instance needs to be reachable from the outside world. So make sure that you have VPC with the security group having a rule to allow ingress traffic on 5432 TCP port.
```

File: examples/postgres-RDS/README.md [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+> You can skip this step if you already have an RDS instance created
+
+RDS instance needs to be reachable from the outside world. So make sure that you have VPC with the security group having the rule to allow ingress traffic on 5432 TCP port.
+
+
+You can create a security group and add rules to the default VPC using the following commands
+
+```bash
+aws ec2 create-security-group --group-name pgtest-sg --description "pgtest security group"
+aws ec2 authorize-security-group-ingress --group-name pgtest-sg --protocol tcp --port 5432 --cidr 0.0.0.0/0
```
Comment (shuguet) [inline examples/postgres-RDS/README.md:29]: Can we add a note to say this is highly insecure and shouldn't be use in a production environment/is only good for testing?
That rule basically makes this database accessible from any source IP in the entire internet ;)

**Review by shuguet — CHANGES_REQUESTED** [review]
This security aspect has to be fixed before release.

**Comment by pavannd1** [comment]
Relevant

**Comment by pavannd1** [comment]
Refactored and merged using a different PR.

**Pull request closed without merging**